R/util.R

```diff
@@ -514,6 +514,107 @@ util.assign_list_to_environment <- function(l, environment = .GlobalEnv){
     }
 }
 
+util.list_files <- function (initial_path, max_depth = 2, current_depth = 0) {
+    # Like base::list.files, but better:
+    # * Lists files, not directories. Because it's called list_files.
+    # * Rather than choosing between no recursion (not very useful) and full
+    #   recursion (potentially very slow for a deep folder tree) you can set a
+    #   max depth. Choose 0 for no recursion.
+    # * Ignores hidden and system files, defined as anything starting with
+    #   '.', '$', or '~'.
+    # * Only works on unix-like systems (no windows)!
+    #
+    # Args:
+    #   initial_path: atomic char, directory to scan for files.
+    #   max_depth: atomic int, default 2, how many subfolders deep to scan for
+    #     files. Zero means enter no subfolders.
+    # current_depth: internal use only, do not specify.
+    #
+    # Returns: char of absolute file paths
+
+    if (.Platform$OS.type != 'unix') {
+        stop("list_files() only supports unix-like systems, not windows.")
+    }
+
+    # List everything within this path, both files and dirs.
+    all_names <- list.files(initial_path, full.names = TRUE)
+
+    # List directories within this path.
+    dirs <- list.dirs(initial_path, recursive = FALSE)
+
+    # Exclude any hidden or system files.
+    ignore_pattern <- '/[\\.\\$~]'
+    all_names <- all_names[!grepl(ignore_pattern, all_names)]
+    dirs <- dirs[!grepl(ignore_pattern, dirs)]
+
+    # Figure out what the real files are by excluding dirs.
+    files <- setdiff(all_names, dirs)
+
+    # If not at max depth, recurse into each found directory.
+    if (current_depth < max_depth) {
+        for (d in dirs) {
+            files <- c(files, list_files(
+                d, max_depth = max_depth, current_depth = current_depth + 1))
```

Comment:
Why is this `list_files` instead of `util.list_files`? I don't see `list_files` defined anywhere...is this some magic that I don't know about yet?